### ISSUE TITLE
Incorrect GRU prediction on small size data

### ISSUE THREAD

**Issue by stas-semenov** [fchollet/keras#8588]
I have an issue with results in examples/image_ocr.py script.

Sample works perfect when I predict data of size equal to minibatch_size (32) or greater.
But if I run prediction of any small size data, e.g. 1, 2, or 3 images, I got absolutely strange and incorrect answers, like random.
Predicting 10-20 samples sometimes look good but with 20-30% errors.
Is it GRU's feature or bug? Or something else?
GRU layers in the example are not stateful and shouldn't depend on minibatch size.

**Comment by stas-semenov** [comment]
I have done some investigation myself.
The issue happens only if add BatchNormalization.

    inner = Dense(get_output_size(), kernel_initializer=ki) (gru_2_conc)
    inner = BatchNormalization() (inner)
    y_pred = Activation('softmax', name='the_activation') (inner)
